src/sql/workbench/contrib/notebook/browser/notebook.component.ts

```diff
@@ -137,6 +137,16 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 	}
 
 	ngOnInit() {
+		/// For focusing the correct cell on tabbing, we hook into the focus_in and check if the active cell
+		/// is parent of the tabbed into element.
+		this._register(DOM.addDisposableListener(this.container.nativeElement, DOM.EventType.FOCUS_IN, () => {
+			const activeCellElement = this.container.nativeElement.querySelector(`.editor-group-container.active .notebook-cell.active`);
+			if (!DOM.isAncestor(document.activeElement, activeCellElement)) {
+				let next = (this.findCellIndex(this.model.activeCell) + 1) % this.cells.length;
```

Comment:
1. Tabbing can go backwards to, so always selecting the next cell isn't going to be correct
2. In general we shouldn't be assuming that focus is going to be following a direct order like this. For example if the active cell is somewhere down below I can still click outside the editor to make the active element something there and then tab until I get into the Notebook editor, at which point the cell I tabbed into will be whatever one happened to be at the top

Reply:
You shouldn't be checking the tab order. Instead you should use the current active element on the document and then loop through the cells until you find the one that is the ancestor of the element being focused. Try just doing that brute force method first and then we can see if it's necessary to figure out some better way perf wise to do it (since that could easily get expensive, although hopefully the focus isn't shifting around *that* much to make a huge difference)

Reply:
Finding the correct cell to focus on using the brute force on elements is okay but we'll not be able to update the model to correct activeCell since there is no way to get that from the html element.